### PR TITLE
School Info Confirmation Dialog - fetch school name for signed in user

### PR DESCRIPTION
**Pull request by nkiruka**
An api end point was added to fetch the school name for a signed in user (teacher).  
Part 2: [PR 27534](https://github.com/code-dot-org/code-dot-org/pull/27534)

### CODE REVIEW

File: apps/src/templates/SchoolInfoConfirmationDialog.jsx [hunk 1]
```diff
@@ -21,24 +21,46 @@ class SchoolInfoConfirmationDialog extends Component {
 
   constructor(props) {
     super(props);
-
     this.state = {
       showSchoolInterstitial: false,
       schoolName: props.schoolName,
```
Comment (bethanyaconnor) [inline apps/src/templates/SchoolInfoConfirmationDialog.jsx:26]: Should this be required? (Github won't let me put this comment comment in propTypes)

File: apps/src/templates/SchoolInfoConfirmationDialog.jsx [hunk 2]
```diff
@@ -21,24 +21,46 @@ class SchoolInfoConfirmationDialog extends Component {
...
+    schoolName: ''
+  };
+
+  componentDidMount() {
+    const {schoolName} = this.state;
+    if (!schoolName && schoolName.length > 0) {
+      fetch('/dashboardapi/v1/users/me/school_name')
+        .then(response => response.json())
+        .then(data => {
+          console.log(data);
```
Comment (bethanyaconnor) [inline apps/src/templates/SchoolInfoConfirmationDialog.jsx]: Debug statement?

File: apps/src/templates/SchoolInfoConfirmationDialog.jsx [hunk 3]
```diff
@@ -21,24 +21,46 @@ class SchoolInfoConfirmationDialog extends Component {
...
 
+  static defaultProps = {
+    onUpdate: () => {},
+    schoolName: ''
+  };
+
+  componentDidMount() {
+    const {schoolName} = this.state;
+    if (!schoolName && schoolName.length > 0) {
+      fetch('/dashboardapi/v1/users/me/school_name')
```
Comment (bethanyaconnor) [inline apps/src/templates/SchoolInfoConfirmationDialog.jsx:39]: Shouldn't this be supplied by the controller?

Comment (nkiruka) [reply apps/src/templates/SchoolInfoConfirmationDialog.jsx:39]: Yes, it is supplied by the controller.  I added a controller method to get the school name

Comment (bethanyaconnor) [reply apps/src/templates/SchoolInfoConfirmationDialog.jsx:39]: Sorry if I've just forgotten a past PR but how do we decide when to show this dialog again? Where is that code?

File: apps/src/templates/SchoolInfoConfirmationDialog.jsx (hunk 1 above)
Comment (nkiruka) [reply apps/src/templates/SchoolInfoConfirmationDialog.jsx:26]: @bethanyaconnor Are you referring to schoolName?  There might be times that a teacher did not enter the name of the school; so I decided to have a default value of an empty string

File: apps/src/templates/SchoolInfoConfirmationDialog.jsx (hunk 3 above)
Comment (nkiruka) [reply apps/src/templates/SchoolInfoConfirmationDialog.jsx:39]: Good question, I sent another PR that includes the code.  I added a helper method that contains the logic for when to show the dialog.

File: apps/src/templates/SchoolInfoConfirmationDialog.jsx [hunk 4]
```diff
@@ -21,24 +21,45 @@ class SchoolInfoConfirmationDialog extends Component {
...
     super(props);
-
     this.state = {
       showSchoolInterstitial: false,
       schoolName: props.schoolName,
       isOpen: true
     };
   }
 
+  static defaultProps = {
```
Comment (epeach) [inline apps/src/templates/SchoolInfoConfirmationDialog.jsx:31]: How do these defaultProps get used?

File: apps/src/templates/SchoolInfoConfirmationDialog.jsx [hunk 5]
```diff
@@ -21,24 +21,45 @@ class SchoolInfoConfirmationDialog extends Component {
...
+    const {schoolName} = this.state;
+    if (!schoolName && schoolName.length > 0) {
+      fetch('/dashboardapi/v1/users/me/school_name')
+        .then(response => response.json())
+        .then(data => {
+          this.setState({
+            schoolName: data.school_name
+          });
+        })
+        .catch(error => this.setState({error}));
```
Comment (epeach) [inline apps/src/templates/SchoolInfoConfirmationDialog.jsx:46]: I'm not familiar with this setState syntax - to what is the error assigned?

File: apps/src/templates/SchoolInfoConfirmationDialog.jsx (hunk 4 above)
Comment (nkiruka) [reply apps/src/templates/SchoolInfoConfirmationDialog.jsx:31]: @epeach The default props will be used unless the parent component over-rides them.

File: apps/src/templates/SchoolInfoConfirmationDialog.jsx (hunk 5 above)
Comment (nkiruka) [reply apps/src/templates/SchoolInfoConfirmationDialog.jsx:46]: If the request to fetch schoolName is unsuccessful, an error is thrown to activate the catch call.

**Comment by nkiruka** [comment]
@epeach @bethanyaconnor Thanks for the code review, let me know if you have any additional questions or if this PR is good to go.

File: apps/src/templates/SchoolInfoConfirmationDialog.jsx (hunk 5 above)
Comment (epeach) [reply apps/src/templates/SchoolInfoConfirmationDialog.jsx:46]: Sorry - my question should have been clearer! Once the catch call is activated, and `this.setState({error})` is called, to which part of the state is `error` set?

File: apps/src/templates/SchoolInfoConfirmationDialog.jsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/templates/SchoolInfoConfirmationDialog.jsx:26]: Makes sense. Thanks!

**Review by bethanyaconnor — APPROVED** [review]

File: apps/src/templates/SchoolInfoConfirmationDialog.jsx (hunk 5 above)
Comment (nkiruka) [reply apps/src/templates/SchoolInfoConfirmationDialog.jsx:46]: Ah I see, it's the local state of the component

File: dashboard/app/controllers/api/v1/users_controller.rb [hunk 6]
```diff
@@ -12,6 +12,13 @@ def load_user
     @user = current_user
   end
 
+  # GET /api/v1/users/<user_id>/school_name
+  def get_school_name
+    render json: {
+      school_name: current_user&.school
+    }
+  end
```
Comment (islemaster) [inline dashboard/app/controllers/api/v1/users_controller.rb:20]: This whole controller strikes me as very odd.  Our route takes a `user_id` parameter, but it _must_ be `me` or the current user's id, and our controller method here ignores the parameter anyway.  Something we'd want to clean up in the future?  At the very least, I'd suggest this method use `@user.school`, and have a test that confirms we can only get the current_user's school name, so if the `load_user` method is changed in the future we know this behavior is preserved.

**Pull request merged**